### ISSUE TITLE
Role is not downgrading rpms

### ISSUE THREAD

**Issue by jason-callaway** [redteam-project/cyber-range-target#3]
Environment

```
[root@lem-test ~]# cat site.yml 
---
- hosts: localhost
  connection: local
  become: true
  roles:
    - cyber-range-target
  vars:
    cves_to_test:
    - CVE-2014-6271
[root@lem-test ~]# ansible-playbook site.yml 
 [WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not
match 'all'


PLAY [localhost] ************************************************************************************************

TASK [Gathering Facts] ******************************************************************************************
ok: [localhost]

TASK [cyber-range-target : Cyber Test Range Target | Install Yum Security Plugin] *******************************
skipping: [localhost]

TASK [cyber-range-target : Cyber Test Range Target | Ensure Custom Facts Directory Exists] **********************
ok: [localhost]

TASK [cyber-range-target : Cyber Test Range Target | Install Custom Fact Module for CVE's] **********************
ok: [localhost]

TASK [cyber-range-target : Cyber Test Range Target | Install Custom Fact Module for CPE] ************************
ok: [localhost]

TASK [cyber-range-target : Cyber Test Range Target | Install Custom Fact Module for Available Packages] *********
ok: [localhost]

TASK [cyber-range-target : Cyber Test Range Target | Obtain CVE Information] ************************************
ok: [localhost] => (item=CVE-2014-6271)

TASK [cyber-range-target : Cyber Test Range Target | Build Package Information] *********************************
ok: [localhost] => (item=CVE-2014-6271 - cpe:/o:redhat:enterprise_linux:5)
ok: [localhost] => (item=CVE-2014-6271 - cpe:/o:redhat:enterprise_linux:6)
ok: [localhost] => (item=CVE-2014-6271 - cpe:/o:redhat:enterprise_linux:7)
ok: [localhost] => (item=CVE-2014-6271 - cpe:/o:redhat:rhel_els:4)
ok: [localhost] => (item=CVE-2014-6271 - cpe:/o:redhat:rhel_eus:5.9)
ok: [localhost] => (item=CVE-2014-6271 - cpe:/o:redhat:rhel_eus:6.4)
ok: [localhost] => (item=CVE-2014-6271 - cpe:/o:redhat:rhel_mission_critical:5.6)
ok: [localhost] => (item=CVE-2014-6271 - cpe:/o:redhat:rhel_mission_critical:6.2)
ok: [localhost] => (item=CVE-2014-6271 - cpe:/a:redhat:rhel_sjis:5)
ok: [localhost] => (item=CVE-2014-6271 - cpe:/a:redhat:rhel_sjis:6)
ok: [localhost] => (item=CVE-2014-6271 - cpe:/a:redhat:rhev_manager:3)

TASK [cyber-range-target : Cyber Test Range Target | Set SELinux State] *****************************************
ok: [localhost]

TASK [cyber-range-target : Cyber Test Range Target | Install Git and Python Virtualenv] *************************
ok: [localhost]

TASK [cyber-range-target : Cyber Test Range Target | Install Appropriate Packages Not Kernel] *******************
skipping: [localhost] => (item=CVE-2014-6271 - cpe:/o:redhat:enterprise_linux:5) 
skipping: [localhost] => (item=CVE-2014-6271 - cpe:/o:redhat:enterprise_linux:6) 
skipping: [localhost] => (item=CVE-2014-6271 - cpe:/o:redhat:enterprise_linux:7) 
skipping: [localhost] => (item=CVE-2014-6271 - cpe:/o:redhat:rhel_els:4) 
skipping: [localhost] => (item=CVE-2014-6271 - cpe:/o:redhat:rhel_eus:5.9) 
skipping: [localhost] => (item=CVE-2014-6271 - cpe:/o:redhat:rhel_eus:6.4) 
skipping: [localhost] => (item=CVE-2014-6271 - cpe:/o:redhat:rhel_mission_critical:5.6) 
skipping: [localhost] => (item=CVE-2014-6271 - cpe:/o:redhat:rhel_mission_critical:6.2) 
skipping: [localhost] => (item=CVE-2014-6271 - cpe:/a:redhat:rhel_sjis:5) 
skipping: [localhost] => (item=CVE-2014-6271 - cpe:/a:redhat:rhel_sjis:6) 
skipping: [localhost] => (item=CVE-2014-6271 - cpe:/a:redhat:rhev_manager:3) 

TASK [cyber-range-target : Cyber Test Range Target | Install Different Kernel Version if Necessary] *************
skipping: [localhost] => (item=CVE-2014-6271 - cpe:/o:redhat:enterprise_linux:5) 
skipping: [localhost] => (item=CVE-2014-6271 - cpe:/o:redhat:enterprise_linux:6) 
skipping: [localhost] => (item=CVE-2014-6271 - cpe:/o:redhat:enterprise_linux:7) 
skipping: [localhost] => (item=CVE-2014-6271 - cpe:/o:redhat:rhel_els:4) 
skipping: [localhost] => (item=CVE-2014-6271 - cpe:/o:redhat:rhel_eus:5.9) 
skipping: [localhost] => (item=CVE-2014-6271 - cpe:/o:redhat:rhel_eus:6.4) 
skipping: [localhost] => (item=CVE-2014-6271 - cpe:/o:redhat:rhel_mission_critical:5.6) 
skipping: [localhost] => (item=CVE-2014-6271 - cpe:/o:redhat:rhel_mission_critical:6.2) 
skipping: [localhost] => (item=CVE-2014-6271 - cpe:/a:redhat:rhel_sjis:5) 
skipping: [localhost] => (item=CVE-2014-6271 - cpe:/a:redhat:rhel_sjis:6) 
skipping: [localhost] => (item=CVE-2014-6271 - cpe:/a:redhat:rhev_manager:3) 

TASK [cyber-range-target : Cyber Test Range Target | Set Fact for Changed Kernel] *******************************
ok: [localhost]

TASK [cyber-range-target : Cyber Test Range Target | Add Script to Remove Unecessary Kernels] *******************
skipping: [localhost]

TASK [cyber-range-target : Cyber Test Range Target | Add One Time Service to Remove Kernels - EL 7] *************
skipping: [localhost]

TASK [cyber-range-target : Cyber Test Range Target | Enable One Time Service to Remove Kernels - EL 7] **********
skipping: [localhost]

TASK [cyber-range-target : Cyber Test Range Target | Restart Host to Effect Kernel Change] **********************
skipping: [localhost]

PLAY RECAP ******************************************************************************************************
localhost                  : ok=10   changed=0    unreachable=0    failed=0   

[root@lem-test ~]# yum update
No packages marked for update
```

We would expect at least one rpm to be listed for update after applying the playbook.

**Comment by jason-callaway** [comment]
Note this issue is different than the old project's issue [#3](https://github.com/fedoraredteam/cyber-range-target/issues/3).

**Comment by jason-callaway** [comment]
Problematic play:

https://github.com/redteam-project/cyber-range-target/blob/6712fc4c7a88d27205f1aeb1426e9b827775b58d/tasks/main.yml#L69-L77

**Comment by jason-callaway** [comment]
Debug information from `ansible-playbook -vvv` run (truncated to relevant snippet):

```
skipping: [localhost] => (item=CVE-2014-6271 - cpe:/o:redhat:enterprise_linux:7)  => {
    "changed": false, 
    "item": {
        "ansible_facts": {
            "cpe": "cpe:/o:redhat:enterprise_linux:7", 
            "cve": "CVE-2014-6271", 
            "package": "bash-4.2.45-5.el7_0.2", 
            "package_name": "bash"
        }, 
        "changed": false, 
        "failed": false, 
        "item": [
            {
                "_ansible_ignore_errors": null, 
                "_ansible_item_label": "CVE-2014-6271", 
                "_ansible_item_result": true, 
                "_ansible_no_log": false, 
                "ansible_facts": {
                    "cve": {
                        "acknowledgement": "\nRed Hat would like to thank Stephane Chazelas for reporting this issue.\n    ", 
                        "bugzilla": {
                            "description": "\nCVE-2014-6271 bash: specially-crafted environment variables can be used to inject shell commands\n    ", 
                            "id": "1141597", 
                            "url": "https://bugzilla.redhat.com/show_bug.cgi?id=1141597"
                        }, 
                        "cs_aw": "True", 
                        "cvss": {
                            "cvss_base_score": "7.5", 
                            "cvss_scoring_vector": "AV:N/AC:L/Au:N/C:P/I:P/A:P", 
                            "status": "verified"
                        }, 
                        "cwe": "CWE-78", 
                        "details": [
                            "\nGNU Bash through 4.3 processes trailing strings after function definitions in the values of environment variables, which allows remote attackers to execute arbitrary code via a crafted environment, as demonstrated by vectors involving the ForceCommand feature in OpenSSH sshd, the mod_cgi and mod_cgid modules in the Apache HTTP Server, scripts executed by unspecified DHCP clients, and other situations in which setting the environment occurs across a privilege boundary from Bash execution, aka \"ShellShock.\"  NOTE: the original fix for this issue was incorrect; CVE-2014-7169 has been assigned to cover the vulnerability that is still present after the incorrect fix.\n    ", 
                            "\nA flaw was found in the way Bash evaluated certain specially crafted environment variables. An attacker could use this flaw to override or bypass environment restrictions to execute shell commands. Certain services and applications allow remote unauthenticated attackers to provide environment variables, allowing them to exploit this issue.\n    "
                        ], 
                        "iava": "2014-A-0142", 
                        "name": "CVE-2014-6271", 
                        "package_state": [
                            {
                                "cpe": "cpe:/a:redhat:rhev_manager:3", 
                                "fix_state": "Affected", 
                                "package_name": "rhev-hypervisor", 
                                "product_name": "RHEV Manager 3"
                            }, 
                            {
                                "cpe": "cpe:/o:redhat:enterprise_linux:3", 
                                "fix_state": "Affected", 
                                "package_name": "bash", 
                                "product_name": "Red Hat Enterprise Linux 3"
                            }, 
                            {
                                "cpe": "cpe:/o:redhat:enterprise_linux:6", 
                                "fix_state": "Affected", 
                                "package_name": "guest-images", 
                                "product_name": "Red Hat Enterprise Linux 6"
                            }, 
                            {
                                "cpe": "cpe:/o:redhat:enterprise_linux:7", 
                                "fix_state": "Affected", 
                                "package_name": "rhel-guest-image", 
                                "product_name": "Red Hat Enterprise Linux 7"
                            }
                        ], 
                        "public_date": "2014-09-24T00:00:00", 
                        "references": [
                            "\nhttps://access.redhat.com/articles/1200223\nhttps://securityblog.redhat.com/2014/09/24/bash-specially-crafted-environment-variables-code-injection-attack\n    "
                        ], 
                        "threat_severity": "Critical"
                    }
                }, 
                "changed": false, 
                "failed": false, 
                "item": "CVE-2014-6271"
            }, 
            {
                "advisory": "RHSA-2014:1293", 
                "cpe": "cpe:/o:redhat:enterprise_linux:7", 
                "package": "bash-4.2.45-5.el7_0.2", 
                "product_name": "Red Hat Enterprise Linux 7", 
                "release_date": "2014-09-24T00:00:00"
            }
        ]
    }, 
    "skip_reason": "Conditional result was False"
}
```

So the problem is in the conditional for the play at:

https://github.com/redteam-project/cyber-range-target/blob/6712fc4c7a88d27205f1aeb1426e9b827775b58d/tasks/main.yml#L74

**Comment by jason-callaway** [comment]
Interesting. That conditional is working. Now looking into [available-packages.fact](https://github.com/redteam-project/cyber-range-target/blob/6712fc4c7a88d27205f1aeb1426e9b827775b58d/files/available_packages.fact).

**Comment by jason-callaway** [comment]
Aaand I as wrong. It was the conditional logic. Fixed by 2d519301c5b25d93257378764b5bae86873565f8
